src/abstractarray.jl

```diff
@@ -127,7 +127,7 @@ When `a` is an instance or a concrete type the element type `eltype(a)` is
 used. However, when `a` is a `UnionAll` type such as `SMatrix{2,2}`, the
 promoted type of `elements` is used instead.
 """
-@inline function _construct_similar(a, s::Size, elements::NTuple{L,ET}) where {L,ET}
+@inline function _construct_similar(a, s::Size, elements::Tuple{ET,Vararg{ET,L}}) where {L,ET}
```

Comment:
Isn't this should be the following?
```suggestion
@inline function _construct_similar(a, s::Size, elements::Tuple{Vararg{ET}}) where ET
```

```julia
julia> (NTuple{L,ET} where {L,ET}) == (Tuple{ET,Vararg{ET,L}} where {L,ET})
false

julia> (NTuple{L,ET} where {L,ET}) == (Tuple{Vararg{ET,L}} where {L,ET})
true

julia> (NTuple{L,ET} where {L,ET}) == (Tuple{Vararg{ET}} where ET)
true
```

Reply:
```julia
julia> NTuple{4,Float64} === Tuple{Float64,Vararg{Float64,3}}
true
```
But, if we don't care about `L`, we may as well drop it from the signature.